js/src/components/lineage/NodeTag.tsx

```diff
@@ -0,0 +1,187 @@
+  isInteractive?: boolean; // if true, allows user to manually fetch row count
+}
+
+export function RowCountTag(
+  {
+    node,
+    isAutoFetching = false,
+    isInteractive = true,
+  }: RowCountTagProps) {
+  const { isLoading, data: rowCount, refetch: invokeRowCountQuery , isFetched, isFetching } = useQuery({
+    queryKey: cacheKeys.rowCount(node.name),
+    queryFn:  () => fetchModelRowCount(node.name),
+    enabled: node.resourceType === "model" && isAutoFetching,
+  });
+
+  function ProcessedRowCountTag(
+    { isLoading, rowCount }: {isLoading: boolean, rowCount: ModelRowCount}) {
+    return (<TagLabel>
+              <SkeletonText isLoaded={!isLoading} noOfLines={1} skeletonHeight={2} minWidth={'30px'}>
+                <ModelRowCount rowCount={rowCount} />
+              </SkeletonText>
+            </TagLabel>);
+  }
+
+  function UnprocessedRowCountTag(
+    { isInteractive, invokeFunction }: {isInteractive:boolean,invokeFunction: () => void}) {
+      if (isInteractive) {
+        return (
+          <IconButton
+            aria-label="Query Row Count"
+            icon={<MdQueryStats />}
+            size="xs"
+            onClick={() => {invokeFunction()}}
+            />
+        );
+      }
+
+      return <Icon
+        as={MdOutlineQuestionMark}
+        />;
+  }
+
+  if (isInteractive === false && isFetched === false && isFetching === false) {
+    // Don't show anything if the row count is not fetched and is not interactive.
+    return null;
+  }
+
+
+  return (
+    <Tooltip hasArrow label={isFetched || isFetching || !isInteractive ?"Number of row":"Query the number of row"}>
+      <Tag>
+        <TagLeftIcon as={FiAlignLeft} />
+        {isFetched || isFetching ? (
+          <ProcessedRowCountTag
+            isLoading={isLoading}
+            rowCount={rowCount}
+          />
+        ) : (
+          <UnprocessedRowCountTag
+            isInteractive={isInteractive}
+            invokeFunction={invokeRowCountQuery}
+          />
+        )}
+      </Tag>
+    </Tooltip>
+  );
+}
+
+export async function fetchRowCountsByNodes(nodes: LineageGraphNode[]) {
+  nodes.forEach((node) => {
+    if (node.resourceType === "model") {
+      console.log("Name", node.name);
+    }
+  });
+}
+
+export function FetchRowCountsButton({ nodes }: { nodes: LineageGraphNode[] }) {
+  const [enabled, setEnabled] = useState<boolean>(false);
+  const [index, setIndex] = useState<number>(0);
+  const name = (index < nodes.length) ? nodes[index].name : "";
+  const { isLoading, isFetched } = useQuery({
+    queryKey: cacheKeys.rowCount(name),
+    queryFn:  () => fetchModelRowCount(name),
+    enabled: enabled,
+  });
+
+  useEffect(() => {
+    if (isFetched) {
+      if (index + 1 < nodes.length) {
+        // TODO: Use BFS to walk though all the changed nodes first.
+        setIndex(index + 1);
+      } else{
+        setEnabled(false);
+      }
+    }
+  }, [isFetched, index, nodes]);
```

Comment:
[Recommendataion]

Here is a bit tricky. In `useQuery`, you have more options to do this

1. useQueries
2. useQueryClient + [setQueryData](https://tanstack.com/query/v4/docs/react/reference/QueryClient#queryclientsetquerydata)

Reply:
I had considered using `useQueries` before. However, the `useQueries` in TanStack Query doesn't provide a way to limit the query tasks. When users try to query lots of models and the warehouse response time is slow, it's very easy to hit query timeout.
I have created a story (RC-95) to enhance the query algorithm. We can enhance the query code in that story.

Reply:
So i prefer to use option 2. Yes. please use another story (RC-95) to enhance it